.circleci/docker/build.sh

```diff
@@ -29,10 +20,42 @@ elif [[ "$image" == *-bionic* ]]; then
   UBUNTU_VERSION=18.04
 elif [[ "$image" == *-focal* ]]; then
   UBUNTU_VERSION=20.04
+elif [[ "$image" == *ubuntu* ]]; then
+  UBUNTU_VERSION="$(echo "${image}" | perl -n -e'/ubuntu(\d+\.\d+)/ && print $1')"
+elif [[ "$image" == *centos* ]]; then
+  CENTOS_VERSION="$(echo "${image}" | perl -n -e'/centos(\d+(\.\d+)?(\.\d+)?)/ && print $1')"
+fi
+
+if [ -n "${UBUNTU_VERSION}" ]; then
+  OS="ubuntu"
+elif [ -n "${CENTOS_VERSION}" ]; then
+  OS="centos"
+else
+  echo "Unable to derive operating system base..."
+  exit 1
+fi
+
+DOCKERFILE="${OS}/Dockerfile"
+if [[ "$image" == *-cuda* ]]; then
+  DOCKERFILE="${OS}-cuda/Dockerfile"
+elif [[ "$image" == *-rocm* ]]; then
+  DOCKERFILE="${OS}-rocm/Dockerfile"
 fi
 
 TRAVIS_DL_URL_PREFIX="https://s3.amazonaws.com/travis-python-archives/binaries/ubuntu/14.04/x86_64"
 
+# verify the given variable was parsed from $image correctly
+function check_variable() {
+  if [ "x$1" = x ]; then
+    echo "missing argument to function check_variable()"
+    exit 1
+  fi
+  if [ "x${!1}" = x ]; then
+    echo "variable '$1' not correctly parsed from image='$image'"
+    exit 1
+  fi
+}
```

Comment:
Does this function just check if the variable is not blank?

Wouldn't it be easier to just use the built in comparison flags like `-z`?

```suggestion
function check_variable() {
  if [[ -z "$1" ]]; then
    echo "missing argument to function check_variable()"
    exit 1
  fi
  if [[ -z "${!1}" ]]; then
    echo "variable '$1' not correctly parsed from image='$image'"
    exit 1
  fi
}
```

Reply:
The intent of `check_variable()` was as much to verify parsing was correct, but also to protect future development of the script in case new variables needed parsing but the caller messed up the syntax.

Reply:
Function was removed from latest commit.